mk/ring.mk

```diff
@@ -32,6 +32,7 @@ RING_SRCS = $(addprefix $(RING_PREFIX), \
   crypto/bn/random.c \
   crypto/bn/shift.c \
   crypto/cipher/e_aes.c \
+  crypto/cpu-arm-linux.c \
```

Comment:
This isn't ideal, and goes against the spirit of the filename...

Upon making this mini-PR I remembered why I put the C where I did: if the C bits are ARM-only with no provision for running tests, then so is all the Rust that uses it. So, to make the Rust (and Rust/C interaction) testable on average joe hardware (i.e. x86 Linux), I made cpu-arm-linux.c, uh, not ARM-specific. What's the right way to do this (where "this" is "make it testable on x86 linux, but in a production build only have this stuff present on ARM")?